go/libraries/doltcore/schema/alterschema/modifycolumn.go

```diff
@@ -51,12 +51,37 @@ func ModifyColumn(
 		newCol.IsPartOfPK = true
 	}
 
-	newSchema, err := replaceColumnInSchema(sch, existingCol.Name, newCol, order)
+	newSchema, err := replaceColumnInSchema(sch, existingCol, newCol, order)
 	if err != nil {
 		return nil, err
 	}
 
-	return updateTableWithModifiedColumn(ctx, tbl, newSchema, newCol)
+	updatedTable, err := updateTableWithModifiedColumn(ctx, tbl, newSchema, newCol)
+	if err != nil {
+		return nil, err
+	}
+
+	if newCol.IsPartOfPK != existingCol.IsPartOfPK {
```

Comment:
Is it truly best to delete them and add a note? In the future, whenever they are added, if the implementor were to delete the comment in anticipation of writing the code and then forget to actually do so, then we'd be left with a bug. This is something we're planning to add at some point in the future, so I'd vote to leave the code to prevent a potential future bug.